packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -137,77 +132,40 @@ function validateSegment(segment: string, file = '') {
 	}
 }
 
-function comparator(a: Item, b: Item) {
-	if (a.isIndex !== b.isIndex) {
-		if (a.isIndex) return isSpread(a.file) ? 1 : -1;
-
-		return isSpread(b.file) ? -1 : 1;
+function comparator(a: RouteData, b: RouteData) {
+	// more specific routes go first
+	if (a.segments.length !== b.segments.length) {
+		return a.segments.length > b.segments.length ? -1 : 1;
 	}
 
-	const max = Math.max(a.parts.length, b.parts.length);
-
-	for (let i = 0; i < max; i += 1) {
-		const aSubPart = a.parts[i];
-		const bSubPart = b.parts[i];
-
-		if (!aSubPart) return 1; // b is more specific, so goes first
-		if (!bSubPart) return -1;
-
-		// if spread && index, order later
-		if (aSubPart.spread && bSubPart.spread) {
-			return a.isIndex ? 1 : -1;
-		}
+	const aIsStatic = a.segments.every((segment) => segment.every((part) => !part.dynamic && !part.spread));
+	const bIsStatic = b.segments.every((segment) => segment.every((part) => !part.dynamic && !part.spread));
 
-		// If one is ...spread order it later
-		if (aSubPart.spread !== bSubPart.spread) return aSubPart.spread ? 1 : -1;
+	// static routes go first
+	if (aIsStatic !== bIsStatic) {
+		return aIsStatic ? -1 : 1;
+	}
 
-		if (aSubPart.dynamic !== bSubPart.dynamic) {
-			return aSubPart.dynamic ? 1 : -1;
-		}
+	const aHasSpread = a.segments.some((segment) => segment.some((part) => part.spread));
+	const bHasSpread = b.segments.some((segment) => segment.some((part) => part.spread));
 
-		if (!aSubPart.dynamic && aSubPart.content !== bSubPart.content) {
-			return (
-				bSubPart.content.length - aSubPart.content.length ||
-				(aSubPart.content < bSubPart.content ? -1 : 1)
-			);
-		}
+	// rest parameters go last
+	if (aHasSpread !== bHasSpread) {
+		return aHasSpread ? 1 : -1;
 	}
 
-	// endpoints are prioritized over pages
-	if (a.isPage !== b.isPage) {
-		return a.isPage ? 1 : -1;
+	// prerendered routes go first
```

Comment:
Go first? Even before the endpoints?

Reply:
Im rewriting all the comments to hopefully make it more clear